CRM/Relationshipblock/Utils/RelationshipBlock.php

```diff
@@ -21,33 +21,47 @@ public static function getExistingRelationships($contactID) {
       'is_active' => 1,
       'contact_id_a' => $contactID,
       'contact_id_b' => $contactID,
-      'return' => ['id', 'relationship_type_id', 'contact_id_a', 'contact_id_b',
-        'contact_id_a.display_name', 'contact_id_b.display_name', 'contact_id_a.is_deceased',
-        'contact_id_b.is_deceased'],
-      'options' => ['limit' => 0, 'sort' => 'relationship_type_id.label_a_b ASC', 'or' => [['contact_id_a', 'contact_id_b']]],
+      'return' => [
+        'id',
+        'relationship_type_id',
+        'contact_id_a',
+        'contact_id_b',
+        'contact_id_a.display_name',
+        'contact_id_b.display_name',
+        'contact_id_a.is_deceased',
+        'contact_id_b.is_deceased',
+        'end_date',
+      ],
+      'options' => [
+        'limit' => 0,
+        'sort' => 'relationship_type_id.label_a_b ASC',
+        'or' => [['contact_id_a', 'contact_id_b']],
+      ],
     ]);
     $ret = [];
     foreach ($existingRelationships['values'] as $rel) {
-      $relationshipType = $displayedRelationships[$rel['relationship_type_id']];
-      $dir = $rel['contact_id_a'] == $contactID ? 'a_b' : 'b_a';
-      list($a, $b) = explode('_', $dir);
-      // Make all bidirectional relationships appear as the A side to make the form processing simpler
-      if ($relationshipType['bi']) {
-        $dir = 'a_b';
+      if (!isset($rel['end_date']) || $rel['end_date'] < date('Y-m-d')) {
```

Comment:
The operator will have to be reversed for that to work correctly. Otherwise, only expired relationships are being displayed.

Reply:
@jensschuppe - good catch! Care to put up a PR for that fix?

Reply:
I do, and it will include settings for whether to exclude expired and also pending (upcoming) relationships, so that this will be configurable, if that's ok within this issue's scope.

Reply:
How about an initial PR to fix the bug, which should be a one-line patch and easy to merge.
Then a follow-up PR with the configuration which may be a little more involved.

Reply:
Sure, see #30 for the bugfix and #29 for the settings.